src/lib/proxy-server.ts

```diff
@@ -35,93 +49,167 @@ async function getSiteByHost( domain: string ): Promise< SiteDetails | null > {
 }
 
 /**
- * Starts the proxy server for sites with custom domains
+ * Common handler for both HTTP and HTTPS requests
  */
-export async function startProxyServer(): Promise< boolean > {
-	if ( isProxyRunning ) return true;
+async function handleProxyRequest(
+	req: http.IncomingMessage,
+	res: http.ServerResponse,
+	isHttps = false
```

Comment:
```suggestion
	isHttps: boolean
```
I don't really see a point in assigning a default value here.